### ISSUE TITLE
Generated doc is broken

### ISSUE THREAD

**Issue by LilianBoulard** [dirty-cat/dirty_cat#330]
Since we moved to numpydoc (#316), the documentation is kind of broken for the parameters types, [for example in the SuperVectorizer](https://dirty-cat.github.io/stable/generated/dirty_cat.SuperVectorizer.html#dirty_cat.SuperVectorizer)
![image](https://user-images.githubusercontent.com/33222654/188840814-c984030f-7989-49f8-a499-bcfeb9c26813.png)

Before that, the resolution was effective, so [we had that](https://output.circle-artifacts.com/output/job/d5c4f7e9-3cf8-4faf-b197-e769be49d585/artifacts/0/doc/generated/dirty_cat.SuperVectorizer.html#dirty_cat.SuperVectorizer)
![image](https://user-images.githubusercontent.com/33222654/188841117-2a2010a9-77bd-4fbc-8f27-221a42fe6fe4.png)

**Comment by GaelVaroquaux** [comment]
Does this resolution work with numpydoc in another project?

**Comment by LilianBoulard** [comment]
So, to answer the question, no I haven't!
Though I've noticed sometimes it works and sometimes it doesn't, depending on the section, and I don't necessarily understand why.
So in the end, by merely tweaking the docstring of the `SuperVectorizer` as an example and following carefully the [style guide](https://numpydoc.readthedocs.io/en/latest/format.html), I was able to get a good result, so it comes down to resolving #345.
